packages/eventual-send/src/handled-promise.js

```diff
@@ -563,14 +564,19 @@ export const makeHandledPromise = () => {
 
   // FIXME: This is really ugly to bypass the type system, but it will be better
   // once we use Promise.delegated and don't have any [[Constructor]] behaviours.
-  /** @type {unknown} */
-  const unknownBaseHandledPromise = baseHandledPromise;
-  HandledPromise = /** @type {typeof HandledPromise} */ (
-    unknownBaseHandledPromise
-  );
-
-  // We cannot harden(HandledPromise) because we're a vetted shim which
-  // runs before lockdown() allows harden to function.  In that case,
-  // though, globalThis.HandledPromise will be hardened after lockdown.
+  // @ts-expect-error cast
+  HandledPromise = baseHandledPromise;
+
+  // We're a vetted shim which runs before `lockdown` allows
+  // `harden(HandledPromise)` to function, but single-level `freeze` is a
+  // suitable replacement because all objects reachable from the result are
```

Comment:
```suggestion
  // suitable replacement because all objects reachable from `HandledPromise` are
```
may as well be explicit.